semgrep/semgrep/target_manager.py

```diff
@@ -65,32 +64,48 @@ def resolve_targets(targets: List[str]) -> Set[Path]:
             for target in targets
         )
 
-    @staticmethod
-    def _parse_output(output: str, curr_dir: Path) -> Set[Path]:
-        """
-            Convert a newline delimited list of files to a set of path objects
-            prepends curr_dir to all paths in said list
-
-            If list is empty then returns an empty set
-        """
-        files: Set[Path] = set()
-        if output:
-            files = set(Path(curr_dir) / elem for elem in output.strip().split("\n"))
-        return files
-
     @staticmethod
     def _expand_dir(
-        curr_dir: Path, language: str, visible_to_git_only: bool
+        curr_dir: Path, language: str, respect_git_ignore: bool
     ) -> Set[Path]:
         """
             Recursively go through a directory and return list of all files with
             default file extention of language
         """
+
+        def _parse_output(output: str, curr_dir: Path) -> Set[Path]:
+            """
+                Convert a newline delimited list of files to a set of path objects
+                prepends curr_dir to all paths in said list
+
+                If list is empty then returns an empty set
+            """
+            files: Set[Path] = set()
+            if output:
+                files = set(
+                    Path(curr_dir) / elem for elem in output.strip().split("\n")
+                )
+            return files
+
+        def run_find(curr_dir: Path, extention: str) -> Set[Path]:
+            """
+                Use 'find' to return set of files with given extension in a directory
+            """
+            output = subprocess.run(
```

Comment:
Can we do something like `[p for p in curr_dir.rglob(f"*.{ext}") if p.is_file()]`? `subprocess` will come with a lot of overhead and potential headaches (what if `find` doesn't exist on this system? is that possible?).

Reply:
With some minimal testing using subprocess with find is about as fast as using rglob: https://github.com/returntocorp/semgrep/pull/946#discussion_r439682901

Currently semgrep-core makes the assumption of find existing so this just moves the code from core to python

Reply:
Sounds like a great opportunity to remove that assumption! If we're writing Python, why not write Python? It's also shorter and easier on the eyes. I try to avoid `subprocess` unless absolutely necessary - the equivalent Python code will likely be easier to debug and play nicer with existing code vs. string parsing `subprocess` output.

Reply:
Fair point. Changing to rglob